core/src/main/java/org/apache/iceberg/rest/RESTJacksonModule.java

```diff
@@ -0,0 +1,174 @@
+
+  private RESTJacksonModule() {
+  }
+
+  public static void registerAll(ObjectMapper mapper) {
+    SimpleModule module = new SimpleModule();
+    module
+        .addSerializer(TableIdentifier.class, new TableIdentifierSerializer())
+        .addDeserializer(TableIdentifier.class, new TableIdentifierDeserializer())
+        .addSerializer(Namespace.class, new NamespaceSerializer())
+        .addDeserializer(Namespace.class, new NamespaceDeserializer())
+        .addSerializer(Schema.class, new SchemaSerializer())
+        .addDeserializer(Schema.class, new SchemaDeserializer())
+        .addSerializer(PartitionSpec.class, new PartitionSpecSerializer())
+        .addDeserializer(PartitionSpec.class, new PartitionSpecDeserializer())
+        .addSerializer(SortOrder.class, new SortOrderSerializer())
+        .addDeserializer(SortOrder.class, new SortOrderDeserializer());
+    mapper.registerModule(module);
+  }
+
+  public static class NamespaceDeserializer extends JsonDeserializer<Namespace> {
+    @Override
+    public Namespace deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return namespaceFromNode(p, jsonNode);
+    }
+  }
+
+  public static class NamespaceSerializer extends JsonSerializer<Namespace> {
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      String[] parts = namespace.levels();
+      gen.writeArray(parts, 0, parts.length);
+    }
+  }
+
+  public static class TableIdentifierDeserializer extends JsonDeserializer<TableIdentifier> {
+    @Override
+    public TableIdentifier deserialize(JsonParser p, DeserializationContext ctxt)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      if (!jsonNode.isObject()) {
+        throw new JsonParseException(p, "TableIdentifier must be an object: " + jsonNode);
+      }
+      ObjectNode objectNode = (ObjectNode) jsonNode;
+      Namespace namespace = namespaceFromNode(p, objectNode.get("namespace"));
+      String name = objectNode.get("name").asText();
+      return TableIdentifier.of(namespace, name);
+    }
+  }
+
+  private static Namespace namespaceFromNode(JsonParser parser, JsonNode jsonNode) throws IOException {
+    if (!jsonNode.isArray()) {
+      throw new JsonParseException(parser, "Namespace must be an array: " + jsonNode);
+    }
+    ArrayNode arrayNode = (ArrayNode) jsonNode;
+    String[] levels = new String[arrayNode.size()];
+    for (int i = 0; i < levels.length; i++) {
+      levels[i] = arrayNode.get(i).asText();
+    }
+    return Namespace.of(levels);
+  }
+
+  public static class TableIdentifierSerializer extends JsonSerializer<TableIdentifier> {
+    @Override
+    public void serialize(
+        TableIdentifier identifier, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeStartObject();
```

Comment:
I wonder if we should keep the consistency here: schema, partition spec and sort order all have their own `Parser` that is Iceberg's bridge for JSON serde. Should we also first have `TableIdentifierParser`, `NamespaceParser` and then have these Jackson specific serializers. This would also help Trino/Presto integration because registering a Jackson serializer is not an option.

Reply:
Not sure where my original comment went, so repeating it here.

I had originally done it that way, but last minute made a change. I was going to just use `TableIdentifierParser`, because `Namespace` is just an array of strings. However, if Trino needs it then that's a different story.

I'll work on adding the `*Parser` back in for `TableIdentifier` (and possibly for `Namespace` as it sounds like Trino will need it).

Reply:
I added `TableIdentifierParser` and a suite of tests for it that are similar to the ones used in the recent `SnapshotsRefParser` PR.

I might move that to its own PR so it can be merged sooner.